src/ds5/ds5-options.cpp

```diff
@@ -388,58 +388,37 @@ namespace librealsense
     void external_sync_mode::set(float value)
     {
         command cmd(ds::SET_CAM_SYNC);
-        cmd.param1 = static_cast<int>(value);
-
-        _hwm.send(cmd);
-        _record_action(*this);
-    }
-
-    float external_sync_mode::query() const
-    {
-        command cmd(ds::GET_CAM_SYNC);
-        auto res = _hwm.send(cmd);
-        if (res.empty())
-            throw invalid_value_exception("external_sync_mode::query result is empty!");
-
-        return (res.front());
-    }
-
-    option_range external_sync_mode::get_range() const
-    {
-        return *_range;
-    }
-
-    external_sync_mode2::external_sync_mode2(hw_monitor& hwm, sensor_base* ep)
-        : _hwm(hwm), _sensor(ep)
-    {
-        _range = [this]()
+        if (_ver == 1)
         {
-            return option_range{ ds::inter_cam_sync_mode::INTERCAM_SYNC_DEFAULT,
-                                 ds::inter_cam_sync_mode::INTERCAM_SYNC_MAX,
-                                 1,
-                                 ds::inter_cam_sync_mode::INTERCAM_SYNC_DEFAULT };
-        };
-    }
-
-    void external_sync_mode2::set(float value)
-    {
-        if (_sensor->is_streaming())
-            throw std::runtime_error("Cannot change Inter-camera HW synchronization mode while streaming!");
-
-        command cmd(ds::SET_CAM_SYNC);
-        if (value < 4)
             cmd.param1 = static_cast<int>(value);
+        }
         else
         {
-            cmd.param1 = 4;
-            cmd.param1 |= (static_cast<int>(value - 3)) << 8;
+            if (_sensor->is_streaming())
+                throw std::runtime_error("Cannot change Inter-camera HW synchronization mode while streaming!");
+
+            if (value < 4)
+                cmd.param1 = static_cast<int>(value);
+            else if (value == 259)
+            {
+                cmd.param1 = 0x00010204;
```

Comment:
Pls document the hard-coded values here and in line 408

Reply:
Added detailed comments.